### ISSUE TITLE
Circular import in matplotlib.pyplot under jupyter

### ISSUE THREAD

**Issue by prefrontalvortex** [ipython/ipython#9852]
When I am in a jupyter notebook, and I try "import matplotlib.pyplot" or "from matplotlib import pyplot as plt", I get:

```
ImportError                               Traceback (most recent call last)
...

/usr/local/lib/python2.7/dist-packages/IPython/core/pylabtools.pyc in configure_inline_support(shell, backend)
    359     except ImportError:
    360         return
--> 361     from matplotlib import pyplot
    362
    363     cfg = InlineBackend.instance(parent=shell)

ImportError: cannot import name pyplot
```
[Full traceback here](http://pastebin.com/txPi9gUe)
[import_matplotlib.pyplot_in_jupyter.txt](https://github.com/ipython/ipython/files/409202/import_matplotlib.pyplot_in_jupyter.txt)

Import works fine when in ipython (command line). Running plots from a module from the command line, also fine. I have tried a variety of techniques:

 * Pip install / uninstall matplotlib, ipython, and jupyter in various order
 * Using pip with --no-cache-dir and/or --ignore-installed
 * Deleting ~/.cache, ~/.ipython and ~/.jupyter
 * Making sure no packages are installed with apt-get, only installed with pip
 * Using apt-get to install python-matplotlib, ipython, and python-jupyter

Rolling back to matplotlib 1.4.3 worked, however this lacks features I need. StackOverflow user recommended rolling back ipykernel to 4.3.1, this fixed the issue, and I am able to use the plotting features I need. 

System info:

```
Linux Mint 18 "Sarah"
Python==2.7.12
ipykernel==4.4.0
ipython==5.0.0
ipython-genutils==0.1.0
ipywidgets==5.2.2
jupyter==1.0.0
jupyter-client==4.3.0
jupyter-console==5.0.0
jupyter-core==4.1.0
notebook==4.2.2
numpy==1.11.1
pip 8.1.2 from /usr/local/lib/python2.7/dist-packages (python 2.7)
```
Output of sys.path in ipython and jupyter (same for both):

```
['',
 '/usr/local/bin',
 '/usr/lib/python2.7',
 '/usr/lib/python2.7/plat-x86_64-linux-gnu',
 '/usr/lib/python2.7/lib-tk',
 '/usr/lib/python2.7/lib-old',
 '/usr/lib/python2.7/lib-dynload',
 '/usr/local/lib/python2.7/dist-packages',
 '/usr/lib/python2.7/dist-packages',
 '/usr/lib/python2.7/dist-packages/PILcompat',
 '/usr/lib/python2.7/dist-packages/gtk-2.0',
 '/usr/lib/python2.7/dist-packages/ubuntu-sso-client',
 '/usr/local/lib/python2.7/dist-packages/IPython/extensions',
 '/home/mm/.ipython']
```

**Comment by minrk** [comment]
This should be fixed in ipykernel 4.4.1, released earlier today.

**Comment by prefrontalvortex** [comment]
Yup! I just upgraded to test 
`%matplotlib`

by itself, and noticed "4.4.1" and wondered if they patched it already, and it was. _thatwaseasy.jpg_
